test/state-component-mocha.js

```diff
@@ -0,0 +1,139 @@
+// state-component-mocha.js
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.should();
+chai.use(chaiAsPromised);
+
+var _ = require('underscore');
+var noflo = require('noflo');
+var stateComponent = require('../components/state-component.js');
+
+describe('promise-component', function() {
+    it("should reject undefined definition", function() {
+        return Promise.resolve().then(stateComponent).should.be.rejected;
+    });
+    it("should reject empty definition", function() {
+        return Promise.resolve({}).then(stateComponent).should.be.rejected;
+    });
+    it("should reject empty inPorts", function() {
+        return Promise.resolve({inPorts:{}}).then(stateComponent).should.be.rejected;
+    });
+    it("should accept array of inPort names", function() {
```

Comment:
I know that is not what this test case is covering.  I was suggesting adding a test case for this.  It is true you have a test that handles a very basic 2 port object scenario (name only).  Was suggesting it might be nice to have one that reflects how most updaters should really be calling this.  As a best practice, I think we want to encourage folks to set the datatype, description, and required flag, but there is no test in this suite that shows that.